.github/workflows/zowe-cli.yml

```diff
@@ -30,45 +75,86 @@ jobs:
       NODE_OPTIONS: --max_old_space_size=4096
 
     if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository) && !contains(github.event.head_commit.message, '[ci skip]')
-
     steps:
     - name: Checkout
       uses: actions/checkout@v2
 
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v2
       with:
         node-version: ${{ matrix.node-version }}
 
-    - name: Use NPM v7
-      id: npm7
-      run: npm install -g npm@^7
+    - name: Get NPM Version
+      id: npm-version
+      run: echo "::set-output name=number::$(npm --version)"
+
+    - name: Use NPM v8
+      id: npm8
+      run: npm install -g npm@^8
 
     - name: Install Node Package Dependencies
       id: install
       run: npm ci
 
+    - name: Use Original NPM Version
+      id: original-npm-version
+      run: npm install -g npm@${{ steps.npm-version.outputs.number }}
+
     - name: Build Source
       id: build
       run: npm run build --if-present
 
+    - name: Build Windows Binary
+      if: matrix.os == 'windows-latest' && github.event.inputs.test-type == "binary"
+      run: |
+        cargo build --verbose ${{ github.event.inputs.binary-type == 'release' && '--release' || '' }} --manifest-path=zowex/Cargo.toml
+        tar -cvzf zowe.tgz -C zowex/target/${{ github.event.inputs.binary-type == 'release' && 'release' || 'debug' }}  zowe.exe
+
+    - name: Build MacOS Binary
+      if: matrix.os == 'macos-latest' && github.event.inputs.test-type == "binary"
+      run: |
+        cargo build --verbose ${{ github.event.inputs.binary-type == 'release' && '--release' || '' }} --manifest-path=zowex/Cargo.toml
+        tar -cvzf zowe.tgz -C zowex/target/${{ github.event.inputs.binary-type == 'release' && 'release' || 'debug' }}  zowe
+
+    - name: Download Prebuilt Linux Binary
+      if: matrix.os == 'ubuntu-latest' && github.event.inputs.test-type == "binary"
+      uses: actions/download-artifact@v2
+      with:
+        name: zowe-linux-latest.tgz
+
+    - name: Archive Binary
+      if: matrix.os != 'ubuntu-latest' && github.event.inputs.test-type == "binary"
+      id: upload-binary
+      uses: actions/upload-artifact@v2
+      with:
+        name: zowe-${{ matrix.os }}.tgz
+        path: zowe.tgz
+
+    - name: Setup Binary in PATH
+      if: github.event.inputs.test-type == "binary"
+      id: setup-binary
+      run: tar -xvzf zowe.tgz -C ./node_modules/.bin --overwrite
```

Comment:
Should ./node_modules/.bin also be the location that the "zowe config daemon" command puts the zowe executable at a customer site?

Reply:
I don't think `./node_modules/.bin` will work for end users since the `where/which zowe` location depends on their installation/configuration of node and npm, e.g. using nvm. 
For example:
- `/root/.nvm/versions/node/v16.12.0/bin/zowe`
- `C:\Users\USER\scoop\apps\nvm\current\nodejs\nodejs\zowe`
- `C:\Users\USER\AppData\Roaming\npm\zowe`

Maybe something along the lines of `"${process.env.ZOWE_CLI_HOME}/_bin"` or `"~/.zowe/_bin")`

Reply:
Thanks. Makes sense. Many customers may not even have the privilege to write to those NodeJS locations when they run the "zowe config daemon" command.

Reply:
Note: 
The location where we place the binary changed from `./node_modules/.bin` to `./__tests__/__resources__/application_instances` becuase of the `TestEnvironment.setUp` function. (see below)
https://github.com/zowe/zowe-cli/blob/06330fa6c50a907363717ee0412d044ab2587843/__tests__/__src__/environment/TestEnvironment.ts#L43